internal/gatewayapi/envoyextensionpolicy.go

```diff
@@ -330,10 +330,14 @@ func (t *Translator) translateEnvoyExtensionPolicyForRoute(
 			if irListener != nil {
 				for _, r := range irListener.Routes {
 					if strings.HasPrefix(r.Name, prefix) {
-						// return 500 and do not configure EnvoyExtensions in this case
+						// if FailOpen is false, return 500 error directly and not forward the request to the backend.
+						// if FailOpen is true, skip the policy application and requests will be processed as if the
+						// policy was not present. https://github.com/envoyproxy/gateway/issues/5905
 						if errs != nil {
-							r.DirectResponse = &ir.CustomResponse{
-								StatusCode: ptr.To(uint32(500)),
+							if !ShouldFailOpen(errs) {
```

Comment:
Can we perhaps add a log here, warning probably, that this is happening? Should we add some sort of informational status condition for this case?

Reply:
Added a log for failopen with translation errors.

We already have status in the policy surfacing the translation errors.